src/postgres-util/src/replication.rs

```diff
@@ -141,3 +142,37 @@ pub async fn drop_replication_slots(
     }
     Ok(())
 }
+
+/// Fetches the active replication timeline ID from an upstream Postgres source.
+///
+/// Will start a replication connection to allow using the IDENTIFY_SYSTEM call
+/// for fetching the timeline ID.
+pub async fn get_timeline_id(
```

Comment:
This extra wrapping function seems extraneous to me––we only need `get_timeline_id_with_client`, though I'd change that function's name to this one